[ CI ] - Automated Unit tests through CI

# Description:

Allow CI to run unit tests automatically on push.

## Problem:

It has been attempted with the following workflow:

```yml
name: Automated STD Unit Testing

on:
  push:
    paths-ignore:
      - "**/README.md"
  pull_request:
  workflow_dispatch:

env:
  CARGO_TERM_COLOR: always
  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

jobs:
  rust-checks:
    name: Rust Checks
    runs-on: ubuntu-latest
    strategy:
      fail-fast: false
      matrix:
        action:
          - command: test
            args: --workspace --exclude focus --features unit-tests
    steps:
      - name: Install Dependencies
        run: sudo apt-get install git wget flex bison gperf python3 python3-pip python3-venv cmake ninja-build ccache libffi-dev libssl-dev dfu-util libusb-1.0-0 -y
      - name: Checkout repository
        uses: actions/checkout@v4
      - name: Setup Rust
        uses: esp-rs/xtensa-toolchain@v1.5
        with:
          default: true
          buildtargets: esp32s3
          ldproxy: false
          version: "1.86.0"
      - name: Enable Nightly
        run: rustup toolchain install nightly --component rust-src
      - name: Enable caching
        uses: Swatinem/rust-cache@v2
      - name: Run command
        run: cargo ${{ matrix.action.command }} ${{ matrix.action.args }}
```

with following errors:
 ```shell
Run cargo test --workspace --exclude focus --features unit-tests
    Updating crates.io index
    Updating crates.io index
 Downloading crates ...
  Downloaded addr2line v0.24.2
  Downloaded adler2 v2.0.0
  Downloaded getopts v0.2.21
  Downloaded miniz_oxide v0.8.3
  Downloaded rustc-demangle v0.1.24
  Downloaded cfg-if v1.0.0
  Downloaded allocator-api2 v0.2.21
  Downloaded memchr v2.7.4
  Downloaded hashbrown v0.15.2
  Downloaded unicode-width v0.1.14
  Downloaded gimli v0.31.1
  Downloaded compiler_builtins v0.1.146
  Downloaded object v0.36.7
  Downloaded libc v0.2.169
   Compiling compiler_builtins v0.1.146
   Compiling core v0.0.0 (/home/runner/.rustup/toolchains/esp/lib/rustlib/src/rust/library/core)
error: failed to run custom build command for `compiler_builtins v0.1.146`

Caused by:
  process didn't exit successfully: `/home/runner/work/focus/focus/target/debug/build/compiler_builtins-df230e9be64ea43e/build-script-build` (signal: 11, SIGSEGV: invalid memory reference)
warning: build failed, waiting for other jobs to finish...
Error: Process completed with exit code 101.
```

# Acceptance Criteria:

- [x] CI is running unit tests automatically for the modules only

--- a/.github/workflows/unit_tests_ci.yml
+++ b/.github/workflows/unit_tests_ci.yml
@@ -20,19 +20,14 @@ jobs:
       matrix:
         action:
           - command: test
-            args: -v --workspace --exclude focus --features unit-tests
+            args: -v -p hl_driver --lib --features unit-tests
     steps:
       - name: Checkout repository
         uses: actions/checkout@v4
-      - name: Setup Rust
-        uses: esp-rs/xtensa-toolchain@v1.5
-        with:
-          default: true
-          buildtargets: esp32s3
-          ldproxy: false
-          version: "1.86.0"
-      - name: Enable caching
-        uses: Swatinem/rust-cache@v2
+      # The .cargo folder contains configuration that do not work with running the test (probably dependency missing)
+      # We only need the normal rust toolchain to test the STD Unit tests
+      - name: Remove .cargo folder 
+        run: rm -rf .cargo
       - name: Run tests
-        # we use the stable toolchain to run the STD unit tests
+        # We use the stable toolchain to run the STD unit tests
         run: cargo +stable ${{ matrix.action.command }} ${{ matrix.action.args }}

--- a/.github/workflows/unit_tests_ci.yml
+++ b/.github/workflows/unit_tests_ci.yml
@@ -20,7 +20,7 @@ jobs:
       matrix:
         action:
           - command: test
-            args: -v -p hl_driver --lib --features unit-tests
+            args: -v --workspace --exclude focus --features unit-tests
     steps:
       - name: Checkout repository
         uses: actions/checkout@v4

--- a/.github/workflows/unit_tests_ci.yml
+++ b/.github/workflows/unit_tests_ci.yml
@@ -1,0 +1,37 @@
+name: Automated STD Unit Testing
+
+on:
+  push:
+    paths-ignore:
+      - "**/README.md"
+  pull_request:
+  workflow_dispatch:
+
+env:
+  CARGO_TERM_COLOR: always
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+  rust-checks:
+    name: Rust Checks
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        action:
+          - command: test
+            args: -v --workspace -p hl_driver --features unit-tests
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Setup Rust
+        uses: esp-rs/xtensa-toolchain@v1.5
+        with:
+          default: true
+          buildtargets: esp32s3
+          ldproxy: false
+          version: "1.86.0"
+      - name: Enable caching
+        uses: Swatinem/rust-cache@v2
+      - name: Run tests
+        run: cargo ${{ matrix.action.command }} ${{ matrix.action.args }}

--- a/.github/workflows/unit_tests_ci.yml
+++ b/.github/workflows/unit_tests_ci.yml
@@ -20,7 +20,7 @@ jobs:
       matrix:
         action:
           - command: test
-            args: -v --workspace -p hl_driver --features unit-tests
+            args: -v --workspace --exclude focus --features unit-tests
     steps:
       - name: Checkout repository
         uses: actions/checkout@v4
@@ -34,4 +34,5 @@ jobs:
       - name: Enable caching
         uses: Swatinem/rust-cache@v2
       - name: Run tests
-        run: cargo ${{ matrix.action.command }} ${{ matrix.action.args }}
+        # we use the stable toolchain to run the STD unit tests
+        run: cargo +stable ${{ matrix.action.command }} ${{ matrix.action.args }}

